### PR TITLE
Add integration tests for login & authentication, users, settings

### PR DESCRIPTION
**Pull request by DoNotSayYes**
**Problem:**

Add integration tests for login & authentication, users, settings

**Solution:**

**Related Issue:**
#388 

**Test plan:**

- [x] 1. Create a user, check the user is created or not
- [x] 2. Update a user's password
- [x] 3. Sign in with local users, using a valid username but an invalid password
- [x] 4. Sign in with local users, using an invalid username but a valid password
- [x] 5. Sign in with local users, using the valid username and password
- [x] 6. Delete a user
- [x] 7. Sign in with token, using an invalid token
- [x] 8. Sign in with token, using a valid token
- [x] 9. Sign in with kubeconfig, using an invalid kubeconfig
- [x] 10. Sign in with kubeconfig, using a valid kubeconfig
- [x] 11. Edit a setting

### CODE REVIEW

File: tests/integration/api/setting_apis_test.go [hunk 1]
```diff
@@ -3,149 +3,67 @@ package api_test
...
-	"github.com/rancher/harvester/pkg/config"
-	"github.com/rancher/harvester/pkg/settings"
+	harvesterv1alpha1 "github.com/rancher/harvester/pkg/apis/harvester.cattle.io/v1alpha1"
 	. "github.com/rancher/harvester/tests/framework/dsl"
-	"github.com/rancher/harvester/tests/framework/fuzz"
 	"github.com/rancher/harvester/tests/framework/helper"
 )
 
-var _ = Describe("verify settings apis", func() {
+var _ = Describe("Verify settings' apis", func() {
```
Comment (gitlawr) [inline tests/integration/api/setting_apis_test.go]: change to `verify settings APIs`
So as to keep consistent with the other codes given that in most cases the description starts in lowercase.
Check the other places in this PR as well.

File: tests/integration/api/setting_apis_test.go [hunk 2]
```diff
@@ -3,149 +3,67 @@ package api_test
...
-			return
-		}
-		for _, item := range settingList.Items {
-			if err := settingController.Delete(item.Name, &metav1.DeleteOptions{}); err != nil {
-				GinkgoT().Logf("failed to delete tested setting %s/%s, %v", item.Namespace, item.Name, err)
-			}
-		}
+	BeforeEach(func() {
 
 	})
```
Comment (gitlawr) [inline tests/integration/api/setting_apis_test.go]: Remove when it is useless.

File: tests/integration/api/setting_apis_test.go [hunk 3]
```diff
@@ -3,149 +3,67 @@ package api_test
...
-			var setting = gout.H{
-				"metadata": map[string]interface{}{
-					"name": settingName,
-					"labels": map[string]string{
-						"test.harvester.cattle.io": "for-test",
-					},
+			By("update api-ui-version's value")
+			resourceVersion = gjson.GetBytes(respBody, "metadata.resourceVersion").Str
+			APIUIVersion := harvesterv1alpha1.Setting{
+				TypeMeta: metav1.TypeMeta{
```
Comment (futuretea) [inline tests/integration/api/setting_apis_test.go:49]: TypeMeta can be removed

File: tests/integration/api/setting_apis_test.go [hunk 4]
```diff
@@ -3,149 +3,67 @@ package api_test
...
-		if err != nil {
-			GinkgoT().Logf("failed to list tested settings, %v", err)
-			return
-		}
-		for _, item := range settingList.Items {
-			if err := settingController.Delete(item.Name, &metav1.DeleteOptions{}); err != nil {
-				GinkgoT().Logf("failed to delete tested setting %s/%s, %v", item.Namespace, item.Name, err)
-			}
-		}
+	BeforeEach(func() {
```
Comment (futuretea) [inline tests/integration/api/setting_apis_test.go]: can be removed

File: tests/integration/api/user_apis_test.go [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+		scaled                   *config.Scaled
+		secretController         ctlcorev1.SecretController
+		serviceAccountController ctlcorev1.ServiceAccountController
+	)
+
+	BeforeEach(func() {
+		scaled = harvester.Scaled()
+		secretController = scaled.CoreFactory.Core().V1().Secret()
+		serviceAccountController = scaled.CoreFactory.Core().V1().ServiceAccount()
+
```
Comment (futuretea) [inline tests/integration/api/user_apis_test.go]: can be removed

File: tests/integration/api/user_apis_test.go [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+			usersAPI = helper.BuildAPIURL("v1", "harvester.cattle.io.users")
+			usersKubeAPI = helper.BuildAPIURL("", "apis/harvester.cattle.io/v1alpha1/users")
+			loginAPI = helper.BuildAPIURL("", "v1-public/auth?action=login")
+		})
+
+		Specify("create a user", func() {
+
+			By("given a username & password")
+			user := harvesterv1alpha1.User{
+				TypeMeta: metav1.TypeMeta{
```
Comment (futuretea) [inline tests/integration/api/user_apis_test.go]: TypeMeta can be removed

File: tests/integration/api/user_apis_test.go [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+		})
+
+		Specify("update a user's password", func() {
+
+			By("update the user's password")
+			updateAPI := fmt.Sprintf("%s/%s", usersAPI, userID)
+			updatedPassword = fuzz.String(8)
+			user := harvesterv1alpha1.User{
+				TypeMeta: metav1.TypeMeta{
```
Comment (futuretea) [inline tests/integration/api/user_apis_test.go:125]: TypeMeta can be removed

**Review by futuretea — APPROVED** [review]
/LGTM

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**